src/evo/cbtx.cpp

```diff
@@ -30,7 +32,15 @@ bool CheckCbTx(const CTransaction& tx, const CBlockIndex* pindexPrev, CValidatio
         return state.Invalid(ValidationInvalidReason::CONSENSUS, false, REJECT_INVALID, "bad-cbtx-payload");
     }
 
-    if (cbTx.nVersion == 0 || cbTx.nVersion > CCbTx::CURRENT_VERSION) {
+    if (cbTx.nVersion == 0 || cbTx.nVersion > CCbTx::CB_CL_SIG_VERSION) {
+        return state.Invalid(ValidationInvalidReason::CONSENSUS, false, REJECT_INVALID, "bad-cbtx-version");
+    }
+
+    if (pindexPrev && !llmq::utils::IsV20Active(pindexPrev) && cbTx.nVersion == CCbTx::CB_CL_SIG_VERSION) {
+        return state.Invalid(ValidationInvalidReason::CONSENSUS, false, REJECT_INVALID, "bad-cbtx-version");
+    }
+
+    if (pindexPrev && llmq::utils::IsV20Active(pindexPrev) && cbTx.nVersion != CCbTx::CB_CL_SIG_VERSION) {
```

Comment:
just a nit:
may be better to write something like that for better readability and avoid code duplication?
```
if (pindexPrev) {
  bool isV20 = llmq::utils::IsV20Active(pindexPrev);
  bool isCbV20 = cbTx.nVersion != CCbTx::CB_CL_SIG_VERSION;
  if (isV20 != isCbV20) return state.Invalid(...., "bad-cbtx-version");
}
return true;
```